numpy/core/src/umath/matmul.c.src

```diff
@@ -0,0 +1,374 @@
+/* -*- c -*- */
+
+#define _UMATHMODULE
+#define _MULTIARRAYMODULE
+#define NPY_NO_DEPRECATED_API NPY_API_VERSION
+
+#include "Python.h"
+
+#include "npy_config.h"
+#include "numpy/npy_common.h"
+#include "numpy/arrayobject.h"
+#include "numpy/ufuncobject.h"
+#include "numpy/npy_math.h"
+#include "numpy/halffloat.h"
+#include "lowlevel_strided_loops.h"
+
+#include "npy_pycompat.h"
+
+#include "npy_cblas.h"
+#include "arraytypes.h" /* For TYPE_dot functions */
+#include <assert.h>
+
+/*
+ *****************************************************************************
+ **                            BASICS                                       **
+ *****************************************************************************
+ */
+
+#define BLAS_ABLE(n, stride, sz) ((stride % sz == 0) && \
+                                  (stride / sz >= n) && \
+                                  (n < NPY_MAX_INT) && \
```

Comment:
can you rename `sz` to `itemsize` here?

This needs a comment, perhaps
```
/**
 * Determine if a non-contiguous axis has strides and length supported by blas.
 *
 * Strides must be multiples of the itemsize, and must not cause segments of the contiguous axis to overlap.
 * BLAS requires sizes to fit within `int`s.
 */
```

Reply:
ok